src/cpp/core/r_util/RProjectFile.cpp

```diff
@@ -1120,6 +1120,21 @@ bool isWebsiteDirectory(const FilePath& projectDir)
    }
 }
 
+// find the website root (if any) for the given filepath
+FilePath websiteRootDirectory(const FilePath& filePath)
+{
+   FilePath dir = filePath.parent();
+   while (!dir.empty())
```

Comment:
Do you think we should also bail out here if we reach the user's home directory?